docs/src/main/asciidoc/neo4j.adoc

```diff
@@ -321,10 +314,10 @@ We also add some exception handling, in case the resource is called with an inva
 public CompletionStage<Response> getSingle(@PathParam("id") Long id) {
     AsyncSession session = driver.asyncSession();
     return session
-        .readTransactionAsync(tx ->
```

Comment:
Not related to the change itself, but rather to the entire method: this is 22 lines for a single get-by-id call :(

I understand it's all async and non-blocking and so on, but e.g. in comparison to Panache where this would be one line essentially, the signal-to-noise ratio seems not great. Is there any way for simplifying this? Perhaps first show imperative, too, before going to async/reactive?

Reply:
Comparing Panache with the raw Driver is like comparing access via Panache to plain JDBC driver. I don’t see the point here.

Reply:
Good point on the comparison, it's not 100% fair. Still the amount of code for this piece of functionality is overwhelming. Anyways, seems my comment isn't really actionable right now, so feel free to ignore it.

Reply:
We have detailed docs on this: https://neo4j.com/docs/driver-manual/4.0-preview/session-api/ This is about choice. 
Actually, most Neo4j users that use Neo4j-OGM (our object mapper) keep on asking how to work around that object mapping. So we think it's in the interest of our (potential) users to make access to the driver as easy as possible and than let the users chose: Either just using imperative (I think that will be good enough in the vast majority of use cases), using an asynchronous connection (available on all Neo4j versions supported by the driver) or going reactive (available on the Neo4j 4.0 and the best choice in a reactive world based on Quarkus).